app/src/main/java/com/android/calendar/event/EditEventView.java

```diff
@@ -1485,12 +1485,14 @@ public void onItemSelected(AdapterView<?> parent, View view, int position, long
         mModel.mCalendarAccountName = c.getString(EditEventHelper.CALENDARS_INDEX_ACCOUNT_NAME);
         mModel.mCalendarAccountType = c.getString(EditEventHelper.CALENDARS_INDEX_ACCOUNT_TYPE);
         // TODO: try to find a similar color within the new event colors before falling back to the calendar color
-        mModel.setEventColor(Arrays.stream(mModel.getCalendarEventColors())
-                .filter(color -> color == mModel.getEventColor())
-                .findFirst()
-                .orElse(mModel.getCalendarColor()));
+        if (mModel.getCalendarEventColors() != null) {
+            mModel.setEventColor(Arrays.stream(mModel.getCalendarEventColors())
+                    .filter(color -> color == mModel.getEventColor())
+                    .findFirst()
+                    .orElse(mModel.getCalendarColor()));
+            setColorPickerButtonStates(mModel.getCalendarEventColors());
+        }
```

Comment:
Maybe add
```
else {
    mModel.setEventColor(mModel.getCalendarColor());
}
```
here to match old behaviour?

Reply:
Done